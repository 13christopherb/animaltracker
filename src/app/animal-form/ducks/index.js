/** @module animal-form */
import animalFormReducer from './reducers';
export {actions as animalFormActions} from './actions';
export {types as animalFormTypes} from './types';
export default animalFormReducer;