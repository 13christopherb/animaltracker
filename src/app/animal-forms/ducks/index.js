/** @module animal-form */
export {types as animalFormTypes} from './types';