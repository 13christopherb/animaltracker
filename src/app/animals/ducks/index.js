import animalsReducer from './Reducers';
export {actions as animalsActions} from './Actions';
export {types as animalsTypes} from './Types';
export default animalsReducer;