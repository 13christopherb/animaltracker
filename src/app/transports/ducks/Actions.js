import * as transportService from './Operations';
import { types} from "./Types";

export const actions = {
    addTransport,
    getTransports,
    deleteTransport
};

function addTransport(transport) {
    return dispatch => {
        dispatch(request(transport));
        return transportService.addTransport(transport)
            .then(res => {
                    dispatch(success(res))
                },
                error=>{
                    dispatch(failure(error))
                });
    };
    function request(transport) {
        return {type: types.ADD_TRANSPORT_REQUEST, transport}
    }

    function success(transport) {
        return {type: types.ADD_TRANSPORT_SUCCESS, transport}
    }

    function failure(error) {
        return {type: types.ADD_TRANSPORT_FAILURE, error}
    }
}

function getTransports() {
    return dispatch => {
        dispatch(request());
        return transportService.getAllTransports()
            .then(
                res => {
                    dispatch(success(res['transports']))
                },
                error => {
                    dispatch(failure(error))
                }
            );
    };
    function request() {
        return {type: types.GET_TRANSPORTS_REQUEST}
    }

    function success(transports) {
        return {type: types.GET_TRANSPORTS_SUCCESS, transports}
    }

    function failure(error) {
        return {type: types.GET_TRANSPORTS_FAILURE, error}
    }
}

function deleteTransport(transport) {
    return dispatch => {
        dispatch(request(transport));
        return transportService.deleteTransport(transport.id)
            .then(
                res => {
                    dispatch(success(transport))
                },
                error => {
                    dispatch(failure(error))
                }
            );
    };
    function request(transport) {
        return {type: types.DELETE_TRANSPORT_REQUEST, transport}
    }

    function success(transport) {
        return {type: types.DELETE_TRANSPORT_SUCCESS, transport}
    }

    function failure(error) {
        return {type: types.DELETE_TRANSPORT_FAILURE, error}
    }
}