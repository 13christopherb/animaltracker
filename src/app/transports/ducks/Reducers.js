import {types} from "./Types";
import _ from "underscore";

const initialState = {
    transports: []
};

export default function transportsReducer(state=initialState, action) {
    switch (action.type) {
        case types.ADD_TRANSPORT_SUCCESS:
            let transports = [...state['transports']];
            transports.push(action.transport);
            return {
                ...state,
                transports: transports
            };
        case types.GET_TRANSPORTS_SUCCESS:
            return {
                ...state,
                transports: action.transports
            };
        case types.DELETE_TRANSPORT_SUCCESS:
            return {
                ...state,
                transports: _.reject(state['transports'], (t) => {
                    return action.transport.id === t.id;
                })
            };
        default:
            return state
    }
}