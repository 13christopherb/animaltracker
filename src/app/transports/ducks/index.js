import transportsReducer from './Reducers';
export {actions as transportActions} from './Actions';
export {types as transportTypes} from './Types';
export default transportsReducer;