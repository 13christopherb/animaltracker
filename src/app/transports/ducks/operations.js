import {apiInstance} from "../../../utils/api";

export const getAllTransports = () =>
    apiInstance.get('/transports').then(res => res.data);

export const deleteTransport = (id) =>
    apiInstance.delete('/transport/'+id);