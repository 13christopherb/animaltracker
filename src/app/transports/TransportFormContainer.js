import React, {Component} from 'react';
import {connect} from 'react-redux';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import {actions} from "./ducks/Actions";
import {TransportForm} from "./TransportForm";

import 'react-datepicker/dist/react-datepicker.css';
import {Select} from "../form_fields/Select";

class TransportFormContainer extends Component {

    constructor(props) {
        super(props);
        this.state = {
            value: '',
            departs: '',
            arrives: '',
            meetTime: this.props.currentTime
        };
        this.handleInputChange = this.handleInputChange.bind(this);
        this.handleDateTimeChange = this.handleDateTimeChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        //this.props.dispatch(actions.fetchAnimals());
    }


    handleInputChange(e) {
        const target = e.target;
        const value = target.value;
        const name = target.name;
        this.setState({
            [name]: value
        });
    }

    handleDateTimeChange(dateTime) {
        this.setState({
            meetTime: dateTime
        });
    }

    handleSubmit(e) {
        e.preventDefault();
        const transport = {
            departs: this.state.departs,
            arrives: this.state.arrives,
            meetTime: this.state.meetTime.format()
        };
        this.props.dispatch(actions.addTransport(transport));
    }

    render() {
        return (
            <div>
               <TransportForm
                   handleSubmit={this.handleSubmit}
                   handleInputChange={this.handleInputChange}
                   handleDateTimeChange={this.handleDateTimeChange}
                   arrives={this.state.arrives}
                   departs={this.state.departs}
                   meetTime={this.state.meetTime}
                   currentTime={this.props.currentTime}

               />
            </div>
        );
    }
}

function mapStateToProps({}, ownProps) {
    return {
        meetTime: ownProps.currentTime
    }
}

export default connect(
    mapStateToProps,
)(TransportFormContainer);