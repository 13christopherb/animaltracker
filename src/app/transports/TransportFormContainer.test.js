import React from 'react';
import {shallow, mount} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import moment from 'moment';
import TransportFormContainer from "./TransportFormContainer";
import {types} from "./ducks/Types";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const initialState = {
    transports: []
};


const store = mockStore(initialState);

describe('<TransportFormContainer />', () => {
    describe('render()', () => {
        let props = {
            currentTime: moment('2017-09-15 09:30:00')
        };
        test('should render the component', () => {
            const wrapper = shallow(<TransportFormContainer {...props} store={store}/>);
            const component = wrapper.dive();
            expect(toJson(component)).toMatchSnapshot()
        })
    });

    describe('Creating new transport', () => {
        let props = {
            currentTime: moment('2017-09-15 09:30:00')
        };
        describe('user inputs are echoed', () => {
            test('departs select is echoed', () => {
                const wrapper = mount(<TransportFormContainer {...props} store={store}/>)
                wrapper.find('select[name="departs"]').simulate('change', {
                    target: {
                        value: 'new text',
                        type: 'test',
                        name: 'departs'
                    }
                });
                expect(wrapper.find('select[name="departs"]').props().value).toEqual('new text');
            });
            test('arrives select is echoed', () => {
                const wrapper = mount(<TransportFormContainer {...props} store={store}/>);
                wrapper.find('select[name="arrives"]').simulate('change', {
                    target: {
                        value: 'new text',
                        type: 'test',
                        name: 'arrives'
                    }
                });
                expect(wrapper.find('select[name="arrives"]').props().value).toEqual('new text');
            });
        });

        test('should prevent default browser submit event', () => {
            const wrapper = mount(<TransportFormContainer {...props} store={store}/>);
            let prevented = false;
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {
                    prevented = true;
                }
            });
            expect(prevented).toBe(true);
        });

        test('should fire ADD_ANIMAL action', () => {
            const wrapper = mount(<TransportFormContainer {...props} store={store}/>);
            const expectedActions = [
                {type: types.ADD_TRANSPORT_REQUEST}
            ];
            wrapper.find('form').simulate('submit', {
                preventDefault: () => {
                    return false;
                }
            });
            expect(store.getActions()[0]['type']).toEqual(expectedActions[0]['type']);
        })
    });

});
