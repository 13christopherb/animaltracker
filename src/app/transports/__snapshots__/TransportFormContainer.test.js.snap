// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransportFormContainer /> render() should render the component 1`] = `
<div>
  <TransportForm
    arrives=""
    currentTime={"2017-09-15T16:30:00.000Z"}
    departs=""
    handleDateTimeChange={[Function]}
    handleInputChange={[Function]}
    handleSubmit={[Function]}
    meetTime={"2017-09-15T16:30:00.000Z"}
  />
</div>
`;
